fdbclient/ClientWorkerInterface.h

```diff
@@ -88,4 +90,41 @@ struct ProfilerRequest {
 	}
 };
 
+struct SetFailureInjection {
+	constexpr static FileIdentifier file_identifier = 15439864;
+	ReplyPromise<Void> reply;
+	struct ThrottleDiskCommand {
+		// how often should the delay be inserted (0 meaning once, 10 meaning every 10 secs)
+		double delayFrequency;
```

Comment:
As written, I think Frequency is a bit confusing of a term as it usually implies a unit of Hz but this is a unit time in seconds which repeats.  Perhaps `delayInterval`?

But taking a step back, I think a `delayProbability` would be a better model, and is typically how simulation makes such decisions for injection of delays or errors.

Reply:
After reading the rest of the code, I see that the feature implemented here is not delaying of read operations by a random amount between delayMin and delayMax, but rather

- When the Throttle command is received, one value in the min/max range is chosen (once, in `throttleFor()` called by `worker()`)
- The chosen value is used as essentially a "completion interval" where all disk operations complete every `delayFrequency` seconds in unison.

I think `completionInterval` would suggest something closer to the actual meaning than `delayFrequency`.

Also, while `ThrottleDiskCommand` and the other 2 supporting classes have `minDelay` and `maxDelay` which imply some sort of random action within a range, as far as I can tell they are only used once, when the command arrives at the worker and `throttleFor()` is called.  It seems like either a single explicit value should be sent in the request or the throttling feature should vary the completion interval over time within the bounds of the min and max (which would then be min / max completion intervals).

Reply:
Updated based on our discussion